lib/charted.js

```diff
@@ -9,16 +9,19 @@ module.exports = function(app, charted_root) {
   if (charted_root === undefined) { charted_root = '/charted/' }
 
   app.use(charted_root + 'get', function (req, res, next) {
-    var get_url = url.parse(req.url, true).query.url
+    var urlObj = url.parse(req.url, true)
 
-    if (!get_url) {
+    if (!urlObj.query.url) {
       res.statusCode = 400
       res.end('Bad Request: no url provided')
-      next()
       return
     }
 
-    request(decodeURIComponent(get_url), function (err, resp, body) {
+    var get_url = urlObj.query.url.match(/:\/\//) ? urlObj.query.url :
+        ('http://' + req.headers.host + '/' +
+         urlObj.query.url.replace(/^\//, ''))
```

Comment:
So here, instead of doing string concatenation and more regular expressions I think you can use `url.format` with a modified object:

```javascript
var urlObj = url.parse("/hello.txt")
urlObj.hostname = "medium.com"
urlObj.protocol = "http:"
url.format(urlObj);
> http://medium.com
```

Also, you should check whether the website is running on https or http.